### PR TITLE
Support PEM PKCS#8 private keys

### LINKED ISSUE #11: Support PEM encoded PKCS#8 private keys
Currently each consumer needs to create the private key instance themself. It would be handy if the `AdobeClientCredentials` would provide some method which creates a `PrivateKey` from a given String in the format outlined at <https://tools.ietf.org/html/rfc7468#section-10>

The code for that is pretty straight-forward actually and one can use an approach like outlined in https://www.baeldung.com/java-read-pem-file-keys#3-get-private-key-from-pem-string.

The only question is whether only RSA keys should be supported or other types (DSA, EC) as well

### PR DESCRIPTION
**Pull request by kwin**
This closes #11

## Description

Provide helper methods to get a private key from a PEM encoded PKCS#8 string

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by kwin** [comment]
If you are fine with the place of this static method I will add a test.

**Comment by bstopp** [comment]
Actually - now that I'm thinking about this i think there are more things to unpack:

- Yes I'm fine with the static method to generate the Key from a string.
- The `Key` property should be changed to `PrivateKey` to be accurately descriptive.
- This class (even though i already merged #13 ) should _not_ implement Serializable. 
  - Here's why: by making this class serializable it will lend itself to being abused in Jenkins and passed between steps, but these _are_ secrets, and with Serializable and not using `hudson.util.Secret` - they will be passed around in clear text.
  - (Won't add Secret fields as that context is only one use of this library, albeit likely/eventually the most popular)

WDYT?

**Comment by kwin** [comment]
@bstopp I changed this PR as you suggested. Please have another look.

File: src/test/java/io/adobe/cloudmanager/AdobeClientCredentialsTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import org.apache.commons.io.IOUtils;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class AdobeClientCredentialsTest {
+
+    @Test
+    public void testGetKeyFromPem() throws IOException, NoSuchAlgorithmException, InvalidKeySpecException {
+        String pem = IOUtils.resourceToString("/keys/private-key-pkcs8-rsa.pem", StandardCharsets.US_ASCII);
```
Comment (bstopp) [inline src/test/java/io/adobe/cloudmanager/AdobeClientCredentialsTest.java:40]: Please generate the key on the fly, or GitHub will raise an issue that private keys are part the code base. Here's an example in this project:

https://github.com/adobe/aio-lib-java-cloudmanager/blob/1515519af147a005727ff4bd72ac2499404ceca4/src/test/java/io/adobe/cloudmanager/impl/IdentityManagementApiImplTest.java#L68-L72

**Review by bstopp — CHANGES_REQUESTED** [review]

File: src/test/java/io/adobe/cloudmanager/AdobeClientCredentialsTest.java (hunk 1 above)
Comment (kwin) [reply src/test/java/io/adobe/cloudmanager/AdobeClientCredentialsTest.java:40]: IMHO there is no Java API to serialize a PrivateKey into PEM PKCS#8 format.
There must be a way to disable the warning in GitHub as this private key does not grant access to anything....

**Pull request merged**